src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -75,9 +75,15 @@ let priv_manager =
        "polyhedra", (module PolyhedraManager: Manager)]
     in
     let domain = (GobConfig.get_string "ana.apron.domain") in
-    match List.assoc_opt domain options with
-    | Some man -> man
-    | None -> failwith @@ "Apron domain " ^ domain ^ " is not supported. Please check the ana.apron.domain setting."
+    let man = (match List.assoc_opt domain options with
+        | Some man -> man
+        | None -> failwith @@ "Apron domain " ^ domain ^ " is not supported. Please check the ana.apron.domain setting.")
+    in
+    (* Set the manager for deserialization (this step is required when apron results are loaded) *)
+    (* As of now, serialization/deserialization only works for octagons *)
+    let module Man = (val man: Manager) in
+    Manager.set_deserialize Man.mgr;
```

Comment:
This looks almost like the manager is calling itself to set itself for deserialization, but the second `Manager` actually refers to `Apron.Manager` I guess?

Also, wouldn't it make more sense for `ApronAnalysis` to set this since it unwraps the module anyway, no? Maybe not the best style to have a `lazy` computation perform side effects like this that just getting the manager will quietly change some Apron global state.

Reply:
Setting the deserialization manager is actually no longer needed, as no deserialization of Apron data structures is done in Goblint itself, but only in the `apronPrecCompare`. There the deserialization manager is already initialized.

Reply:
So I could remove the code in question entirely.